addons/mass_mailing/static/tests/mass_mailing_favourite_filter_tests.js

```diff
@@ -407,5 +407,72 @@ QUnit.module('favorite filter widget', (hooks) => {
         assert.isNotVisible(fixture.querySelector('.o_mass_mailing_remove_filter'),
             "should not have option to remove filter because mailing domain is changed");
     });
+
+    QUnit.test('mailing_model_name can be empty without throwing an error', async (assert) => {
+        assert.expect(1);
+
+        serverData.models.partner = {
+            fields: {
+                name: { string: 'Name', type: 'char', searchable: true },
+            },
+            records: [
+                { id: 1, name: 'Azure Interior' },
+                { id: 2, name: 'Deco Addict' },
+                { id: 3, name: 'Marc Demo' },
+            ]
+        };
+
+        serverData.models['mailing.filter'].records = [{
+            id: 1,
+            name: 'Azure Partner Only',
+            mailing_domain: "[['name','=', 'Azure Interior']]",
+            mailing_model_id: 2,
+        }];
+
+        serverData.models['mailing.mailing'].records.push({
+            id: 3,
+            display_name: 'Partner Event promotion',
+            subject: 'Early bird discount for Partners!',
+            mailing_model_id: 2,
+            mailing_model_name: '',
+            mailing_filter_count: 1,
+            mailing_domain: "[['name','!=', 'Azure Interior']]",
+            state: 'running',
+        });
+
+        serverData.models['mailing.mailing'].fields.state = {
+            string: 'Stage',
+            type: 'selection',
+            selection: [['draft', 'Draft'], ['running', 'Running']]
+        };
+
+        serverData.models['mailing.mailing'].onchanges = {
+            mailing_filter_id: obj => {
+                obj.mailing_domain = serverData.models['mailing.filter'].records.filter(r => r.id === obj.mailing_filter_id)[0].mailing_domain;
+            },
+        };
+
+        await makeView({
+            type: "form",
+            resModel: "mailing.mailing",
+            resId: 3,
+            serverData,
+            arch: `<form>
+                    <field name="display_name"/>
+                    <field name="subject"/>
+                    <field name="mailing_model_name" invisible="1"/>
+                    <field name="mailing_model_id" attrs="{'readonly': [('state', '!=', 'draft')]}"/>
+                    <field name="mailing_filter_count" />
+                    <field name="mailing_filter_id" widget="mailing_filter" options="{'no_create': '1', 'no_open': '1', 'domain_field': 'mailing_domain', 'model': 'mailing_model_id'}"/>
+                    <field name="state" invisible="1"/>
+                    <group>
+                        <field name="mailing_domain" widget="domain" options="{'model': 'mailing_model_name'}"/>
+                    </group>
+                </form>`,
+        });
+
+        await testUtils.nextTick();
+        assert.containsOnce(fixture, ".o_mass_mailing_save_filter_container");
```

Comment:
Check that it works in edit mode, then in readonly.
Note that checking '.o_mass_mailing_save_filter_container' is not sufficient, you also need to check that it is visible to the user (the parent div is hidden, but the container always exists)
```suggestion
        await testUtils.nextTick();
        const selectField = fixture.querySelector("div[name='state'] select");

        assert.containsOnce(fixture, "div[name='mailing_model_id']:not(.o_readonly_modifier)");
        assert.ok(fixture.querySelector(".o_mass_mailing_save_filter_container").checkVisibility());

        // set to 'running'
        selectField.selectedIndex = 2;
        selectField.dispatchEvent(new Event('click'));
        selectField.dispatchEvent(new Event('change'));
        await testUtils.nextTick();

        assert.containsOnce(fixture, "div[name='mailing_model_id'].o_readonly_modifier");
        assert.ok(fixture.querySelector(".o_mass_mailing_save_filter_container").checkVisibility());
```